modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/ExtensionErrorsModelEnricher.java

```diff
@@ -0,0 +1,126 @@
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypeProvider;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypes;
+import org.mule.runtime.extension.api.annotation.error.Throws;
+import org.mule.runtime.extension.api.declaration.DescribingContext;
+import org.mule.runtime.extension.api.declaration.spi.ModelEnricher;
+import org.mule.runtime.extension.api.error.ErrorTypeDefinition;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.extension.api.model.property.ConnectivityModelProperty;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.ExtensionElement;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.MethodElement;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.runtime.ExtensionTypeWrapper;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.runtime.MethodWrapper;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingMethodModelProperty;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingTypeModelProperty;
+import org.mule.runtime.module.extension.internal.util.IdempotentDeclarationWalker;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * {@link ModelEnricher} implementation which enriches the {@link ExtensionModel} and their {@link OperationModel}
+ * from the used {@link ErrorTypes} and {@link Throws} in an Annotation based extension.
+ *
+ * @since 4.0
+ */
+public class ExtensionErrorsModelEnricher implements ModelEnricher {
+
+  private ExtensionErrorModelFactory errorModelDescriber;
+
+  @Override
+  public void enrich(DescribingContext describingContext) {
+    ExtensionDeclaration declaration = describingContext.getExtensionDeclarer().getDeclaration();
+    Optional<ImplementingTypeModelProperty> implementingType = declaration.getModelProperty(ImplementingTypeModelProperty.class);
+
+    if (implementingType.isPresent()) {
+
+      ExtensionElement extensionElement = new ExtensionTypeWrapper<>(implementingType.get().getType());
+
+      extensionElement.getAnnotation(ErrorTypes.class).ifPresent(errorTypesAnnotation -> {
+
+        String extensionNamespace = getExtensionsErrorNamespace(declaration);
+        ErrorTypeDefinition<?>[] errorTypes = (ErrorTypeDefinition<?>[]) errorTypesAnnotation.value().getEnumConstants();
+
+        if (errorTypes.length > 0) {
+          errorModelDescriber = new ExtensionErrorModelFactory(errorTypes, extensionNamespace);
+          errorModelDescriber.getErrorModels().forEach(declaration::addErrorModel);
+
+          new IdempotentDeclarationWalker() {
+
+            @Override
+            public void onOperation(WithOperationsDeclaration owner, OperationDeclaration declaration) {
```

Comment:
As we talked, we decided to add connection provider support as a future improvement